build/generate-type-id-consts.php

```diff
@@ -0,0 +1,119 @@
+<?php
+
+/*
+ *
+ *  ____            _        _   __  __ _                  __  __ ____
+ * |  _ \ ___   ___| | _____| |_|  \/  (_)_ __   ___      |  \/  |  _ \
+ * | |_) / _ \ / __| |/ / _ \ __| |\/| | | '_ \ / _ \_____| |\/| | |_) |
+ * |  __/ (_) | (__|   <  __/ |_| |  | | | | | |  __/_____| |  | |  __/
+ * |_|   \___/ \___|_|\_\___|\__|_|  |_|_|_| |_|\___|     |_|  |_|_|
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Lesser General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * @author PocketMine Team
+ * @link http://www.pocketmine.net/
+ *
+ *
+ */
+
+declare(strict_types=1);
+
+namespace pocketmine\build\generate_type_id_consts;
+
+/*
+ * This script patches BlockTypeIds and ItemTypeIds with updated constants generated from VanillaBlocks and VanillaItems
+ * respectively.
+ */
+
+use pocketmine\block\Block;
+use pocketmine\block\VanillaBlocks;
+use pocketmine\item\Item;
+use pocketmine\item\ItemBlock;
+use pocketmine\item\VanillaItems;
+use pocketmine\utils\AssumptionFailedError;
+use pocketmine\utils\Filesystem;
+use pocketmine\utils\Utils;
+use function array_filter;
+use function array_map;
+use function asort;
+use function dirname;
+use function max;
+use function preg_last_error_msg;
+use function preg_replace;
+use const SORT_NUMERIC;
+
+require dirname(__DIR__) . '/vendor/autoload.php';
+
+/**
+ * @param int[]    $stringToTypeIdMap
+ * @param string[] $backwardsCompatibilityConstants
+ *
+ * @phpstan-param array<string, int> $stringToTypeIdMap
+ * @phpstan-param array<string>      $backwardsCompatibilityConstants
+ */
+function patchTypeIds(string $file, array $stringToTypeIdMap, array $backwardsCompatibilityConstants) : void{
```

Comment:
What's the point of getting a TypeIdMap here, the name isn't enough?

Reply:
The values are generated by VanillaBlocks, so we need to receive them here to generate the constant values

Reply:
Oh yes, to order them, I've got it